text/0000-lwc-signals.md

```diff
@@ -0,0 +1,372 @@
+1. __Deprecation Warnings__: Introduce deprecation warnings for `@track` and `@wire` in development mode to inform developers of the upcoming changes.
+
+Keep in mind that existing technology can still work side by side with the new reactivity model. For example, `@wire` and `@track` can still be used in the same component with Signals and stores, allowing developers to migrate incrementally.
+
+### Support and Migration
+
+1. __Migration Tools and Assistance__: Provide tools and resources to help developers migrate existing components to the new model. Potentially leveraging generative AI for that.
+1. __Fallback Mechanisms__: Maintain legacy `@track` and `@wire` functionality to ensure that applications continue to function during the transition.
+1. __Long-term Support__: Offer long-term support for the previous reactivity model to give enterprise users sufficient time to plan and execute their migration strategies. The `@wire` and `@track` will certainly get less performant, so developers are encouraged to migrate to the new model.
+
+### Monitoring and Feedback
+1. Feedback Channels: Establish clear channels for feedback from developers to monitor the adoption process's success and identify any areas needing additional support or clarification.
+1. Performance Metrics: Track performance metrics before and after adopting the new model to validate its benefits and guide further optimizations.
+
+## Unresolved Questions
+
+[List any questions or concerns that you have not yet addressed.]
+
+## Security Considerations
+
+Minor security considerations:
+
+1. Changing a property of a value from a store has no implications for other components with access to the same store, but we loose the read-only protection that we get with `@track`. This is not a big deal, but it is something to consider. The responsability shifts to the store developer to protect against others trying to make changes to it. LDS is doing some of that already by producing a new object whenever someone subscribe, but the fact that components are accessing the `.value` directly, makes it more difficult to prevent someone from changing the object that eventually someone else would rely on, unless a new value is emitted.
+2. Exposing more internal implementation details to developers, like the `$api` property, could lead to misuse and potential security issues as described in 1 if the value is an object. An alternative here is to continue doing the readonly proxy for non-primitive values, but that would be a step back in terms of performance.
+
+## Open Discussion Topics
+
+* The signal/store protocol. Except for the `.value`, which seems to be a requirement, the rest of the protocol is open for discussion.
+* The `this.$api` as a way to access internal signal objects created by LWC, that seems to be a good compromise, but we can discuss alternatives.
```

Comment:
It would be great to have examples from other frameworks in this document. I know nearly every framework is moving to signals these days, but I'm not familiar with the exact API patterns used in each one.

Reply:
Agreed 100%. I'm going to attempt to put together a survey of signals implementations & alternate approaches across web frameworks ahead of our dev sync discussion.